.github/workflows/build_and_test.yml

```diff
@@ -0,0 +1,262 @@
+              -Dsonar.login="${{ secrets.SONAR_TOKEN }}"
+
+  tests:
+    needs: build
+    runs-on: ubuntu-latest
+    container:
+      image: openjdk:8-jdk
+    defaults:
+      run:
+        working-directory: .
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Setup System Tools
+        run: |
+          apt update -y
+          apt install -y curl
+      - name: Install Node.js
+        run: |
+          curl -fsSL https://deb.nodesource.com/setup_16.x | bash -
+          apt-get install -y nodejs
+      - name: Check Node.js version
+        run: node --version
+
+      - uses: actions/cache@v4
+        name: Restore gradle cache
+        with:
+          path: |
+            .gradle/caches
+            gradle/wrapper
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+
+      - name: Get Gradle jar
+        run: |
+          if [ ! -f ./gradle/wrapper/gradle-wrapper.jar ]; then
+            ./configure.sh
+          fi
+
+      - name: Unit tests
+        run: |
+          chmod +x ./gradlew
+          ./gradlew --no-daemon --stacktrace test
+      - name: Integration tests
+        run: |
+          chmod +x ./gradlew
+          ./gradlew --no-daemon --stacktrace integrationTest
+
+      - name: Save test results
+        run: |
+          mkdir -p ~/junit/
+          find rskj-core/build/test-results -type f -name "*.xml" -exec cp {} ~/junit/ \;
+        if: always()
+
+  mining-tests:
+    needs: build
+    runs-on: ubuntu-latest
+    services:
+        bitcoind1:
+            image: ghcr.io/rsksmart/rskj/mit_bitcoind1:latest
+            ports:
+            - 8331:8331
+            - 31591:31591
+            - 32591:32591
+            options: --name bitcoind1
+        bitcoind2:
+            image: ghcr.io/rsksmart/rskj/mit_bitcoind2:latest
+            ports:
+            - 8332:8332
+            - 31592:31592
+            - 32592:32592
+            options: --name bitcoind2
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: '12.x'
+      - name: Check Node.js version
+        run: node --version
+
+      - name: Checkout Mining Integration Tests Repository
+        uses: actions/checkout@v4
+        with:
+          repository: rsksmart/mining-integration-tests
+          ref: ${{ secrets.MINING_INTEGRATION_TESTS_REF }}
+          token: ${{ secrets.GITHUB_TOKEN }}
+          path: mining-integration-tests
+
+      - name: Install Mining Integration Tests dependencies
+        working-directory: mining-integration-tests
+        run: |
+          npm ci
+
+      - name: Change mining-integration-tests bitcoind url in config.json to localhost
+        working-directory: mining-integration-tests
+        run: |
+          sed -i 's/bitcoind01/localhost/g' config.json
```

Comment:
I'm not a big fan of this kind of patterns. Can we add an `rskj-ci.json` to rsksmart/mining-integration-tests and here run a `cp -f rskj-ci.json config.json` instead?

Reply:
I mean, sure ...

Reply:
Or perhaps you'd be ok with `jq`-ing it instead?

Reply:
My worries with `sed` and `jq` is that:

- the final content isn't super clear
- it's error prone
- it's hard to maintain

By pushing the CI config into MIT repo, my worries are solved and the behaviour is more predictable.

Put into another way, treat both repositories as objects of different classes. Using `sed` to set private properties of the other object breaks encapsulation.

Reply:
I don't think this analogy works tbh because config files would be public methods, since they are meant to be used by the consumer; otherwise they could have been hardcoded. Which, in the case of the MIT, is the repo that uses it to test itself.
But I'm happy to create a PR in the MIT project for one config file that doesn't change anything for the repo itself. ;)

Reply:
also in MIT README file is:

> 3. Modify `config.json` to match the corresponding url, port and credentials for the bitcoind and rsk node instances

Reply:
as to the list of worries. I get that for `sed`, but with `jq` we can be very precise in what it's done to the json file. It's no different from you manually changing the config were you to run it locally.

Reply:
Ok, go with `jq`.

Reply:
Nevertheless, here you go: https://github.com/rsksmart/mining-integration-tests/pull/32. After you(?) merge this MIT PR, I can adapt the behaviour of this gha job.